core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/utilities/Table.java

```diff
@@ -28,6 +29,75 @@
 
     Set<Integer> separators = new HashSet<>();
 
+    /**
+     * This function creates a text-formatted table
+     * @return String
+     */
+    public String toTextTable() {
+        StringBuilder buffer = new StringBuilder();
+
+        int numCols = this.getColumnCount();
+        int numRows = this.getRowCount();
+
+        List<String> formatterList = new ArrayList<>();
+        List<ArrayList<String>> allTableRows = new ArrayList<>();
+        ArrayList<String> currTableRow = new ArrayList<>();
+
+        // This is used to determine the space between  two '|' in the table
+        int longestWordSize = 10;
+
+        for (int i = 0; i < 2 * numCols + 1; i++) {
+            if (i % 2 == 0) {
+                formatterList.add("%-2s ");
+                currTableRow.add("|");
+            }
+            else {
+                longestWordSize = Math.max(this.getColumnName(i/2).length(), longestWordSize);
+                formatterList.add("%-10s ");
+                currTableRow.add(this.getColumnName(i/2));
+            }
+        }
+        allTableRows.add(currTableRow);
+        formatterList.set(formatterList.size()-1, "%-2s%n");
+
+        for (int row = 0; row < numRows; row++) {
+            currTableRow = new ArrayList<>();
+            for (int i = 0; i < numCols * 2 + 1; i++) {
+                if (i % 2 == 0) {
+                    currTableRow.add("|");
+                }
+                else {
+                    longestWordSize = Math.max(this.getValueAt(row, i/2).toString().length(), longestWordSize);
+                    currTableRow.add(this.getValueAt(row, i/2).toString());
+                }
+            }
+            allTableRows.add(currTableRow);
+        }
+
+        // Change the space between '|' if the longest word has length longer than 10(default)
+        if (longestWordSize > 10) {
+            for (int i = 1; i < formatterList.size(); i += 2) {
+                formatterList.set(i, "%-" + longestWordSize + "s ");
```

Comment:
You should add to the documentation that all columns are of equal size = Max(10, longest word)